pkg/kube/cluster-init.sh

```diff
@@ -54,6 +54,44 @@ wait_for_default_route() {
   return 1
 }
 
+wait_for_vault() {
+        # Wait for vault to unseal
+        vaultMgrStatusPath="/run/vaultmgr/VaultStatus/Application Data Store.json"
+        vaultMgrStatus=0
+        DataSecAtRestStatus_DATASEC_AT_REST_DISABLED=1
+        DataSecAtRestStatus_DATASEC_AT_REST_ENABLED=2
```

Comment:
It would be more maintainable if we didn't need to but this in the script. Don't know if it is worth the effort to implement an inline command in zedbox called waitforvaultready which would do this using pubsub and not need to hard-code paths and constants.

Reply:
@shjala @eriknordmark vaultmgr is already plumbed with support for inline commands so it would be fairly quick to add another for "waitunsealed"/"checkunsealed" or something similar.  The kube container doesn't currently base off dom0-ztools but it is possible to reach over to pillar and use zfs-get like this currently:

`42b04062-e53d-4c17-bbeb-9dc4e6b61f6b:/# LD_LIBRARY_PATH=/hostfs/containers/services/pillar/rootfs/usr/lib/ /hostfs/containers/services/pillar/rootfs/usr/sbin/zfs get -H -o value keystatus persist/vault
available`

Is there is a preference towards one implementation or the other?  If zfs is preferred I could base kube off dom0-ztools to avoid the LD_LIBRARY_PATH.

Reply:
I think doing a waitunsealed command in vaultmgr makes sense.